account_move_line_search_extension/account.py

```diff
@@ -47,7 +47,8 @@ def search(self, cr, uid, args, offset=0, limit=None, order=None,
         if context and 'account_move_line_search_extension' in context:
             ana_obj = self.pool['account.analytic.account']
             for arg in args:
-                if arg[0] == 'analytic_account_id':
+                if arg[0] == 'analytic_account_id' and\
+                   isinstance(arg[0], basestring):
```

Comment:
Well, it's not my personal preference, but a general Python guideline (well, it's more to use brackets than the indentation itself, but one comes after the other), and also for avoiding interpretation problems on nested if: https://stackoverflow.com/a/181557